interfaces/apparmor/backend.go

```diff
@@ -548,6 +571,22 @@ func (b *Backend) Remove(snapName string) error {
 	return errUnload
 }
 
+func (b *Backend) RemoveLate(snapName string, rev snap.Revision, typ snap.Type) error {
+	logger.Noticef("remove late for snap %v (%s) type %v", snapName, rev, typ)
+	if typ != snap.TypeSnapd {
+		// late remove is relevant only for snap confine profiles
+		return nil
+	}
+
+	globs := []string{snapConfineProfileName(snapName, rev)}
+	_, removed, errEnsure := osutil.EnsureDirStateGlobs(dirs.SnapAppArmorDir, globs, nil)
+	errUnload := unloadProfiles(removed, apparmor_sandbox.CacheDir)
```

Comment:
Can you add a `XXX: unloadProfiles() relies on parsing of the original profile definitions to know which profile names to remove. This is not really removing the profiles from memory.`